### ISSUE TITLE
How to move up & down an entry inside Telescope?

### ISSUE THREAD

**Issue by wojciech-kulik** [nvim-telescope/telescope.nvim#3365]
### Description

(I tried reaching out on Gitter first, but it seems like it's abandoned)

I need to add shortcuts to move telescope results up & down. However, I'm struggling with glitches, because replacing results moves the selection to the top, so the list jumps back & forth. Setting selected item without a small delay doesn't work. :( Is there any other solution?

### Neovim version

```markdown
NVIM v0.11.0-dev-1161+g6ef80eb42c
Build type: RelWithDebInfo
LuaJIT 2.1.1731601260
```

### Operating system and version

macOS 15.1

### Telescope version / branch / rev

main

### checkhealth telescope

```markdown
OK
```

### Steps to reproduce

```lua
local function update_results(results)
  if activePicker then
    activePicker:refresh(
      telescopeFinders.new_table({
        results = results,
      }),
      {
        new_prefix = telescopeConfig.prompt_prefix,
      }
    )
  end
end

vim.keymap.set({ "n", "i" }, "<M-y>", function()
  local entries = get_entries()
  local currentEntry = actionState.get_selected_entry()
  if currentEntry then
    local index = currentEntry.index
    if index == 1 then
      return
    end

    swap_entries(entries, index, index - 1)
    save_entries()

    update_results(entries)
    if activePicker then
      activePicker:set_selection(index - 2)
    end
  end
end, { buffer = bufnr })
```

### Expected behavior

The shortcut should move the item up & down without scrolling and jumping

### Actual behavior

It seems like `set_selection` doesn't work directly after `refresh`. Is there any other way to move up & down selected entry without glitches?

### Minimal config

```Lua
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs { "config", "data", "state", "cache" } do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.uv.fs_stat(lazypath) then
  vim.fn.system {
    "git",
    "clone",
    "--filter=blob:none",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  }
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  {
    "nvim-telescope/telescope.nvim",
    dependencies = {
      "nvim-lua/plenary.nvim",
    },
    config = function()
      -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
      require("telescope").setup {}
    end,
  },
}

require("lazy").setup(plugins, {
  root = root .. "/plugins",
})
```

**Comment by Conni2461** [comment]
But why would need to do that? It seems like you want to manully sort (move around) the resilts list.

I cant imagine a usecase where you want to do that

**Comment by wojciech-kulik** [comment]
I use Telescope in my plugin to present to the user a list of his devices with the ability to reorder the list and delete some devices - basically to manage the list.

**Comment by Conni2461** [comment]
Yeah but is telescope the correct solution for your problem or wouldnt it be better to build a ui yourself? Because we are primarly a fuzzy finder over lists and i dont see us implementing bindings or actions to manipulate the entymanager

**Comment by wojciech-kulik** [comment]
The same question could be asked for any other integration, Telescope has been used for many creative things :). Fuzzy finder is helpful to search devices, that's why I use it. Also, I use it for many other things, so it makes sense to keep my plugin consistent. 

I understand that it might be out of the scope of Telescope to support it. I'm just asking if there is any better way to achieve that in Telescope, but based on your answers I assume there isn't. So the only remaining question is why `set_selection` doesn't work after  `refresh` function and is it an expected behavior? (btw. I worked around it in my plugin by adding a small delay)
